### PR TITLE
Speed up partial update with blit operation

### PR DESCRIPTION
**Pull request by erikcorry**

### CODE REVIEW

File: src/pixel_display.toit [hunk 1]
```diff
@@ -272,27 +274,48 @@ abstract class PixelDisplay:
...
   update_frame_buffer clean/bool refresh_dimensions:
-    width := min width_ 128
+    width := min width_ 120
     max_height := round_down (max_canvas_height_ width) 8
 
+    // Outer loop - the coarse rectangles that are the max size of
+    // update patches.
     for y:= 0; y < height_; y += max_height:
       while line_is_clean_ y:
         y += 8
```
Comment (floitsch) [inline src/pixel_display.toit:288]: Why is there a `+= 8`?

File: src/pixel_display.toit [hunk 2]
```diff
@@ -272,27 +274,48 @@ abstract class PixelDisplay:
...
         bottom := min y + max_height height_
+
+        // Quick check if the whole rectangle is clean.  This is a little
+        // imprecise because the same mask is used for all lines.
+        mask := 0
+        for iy := top; iy < bottom; iy += 8:
+          mask |= 1 << ((iy >> 3) & 0b111)
+        idx := (top >> 6) * dirty_bytes_per_line_ + (left >> 3)
+        end_idx := ((round_up bottom 64) >> 6) * dirty_bytes_per_line_
+        dirty_accumulator_[0] = CLEAN_
```
Comment (floitsch) [inline src/pixel_display.toit:304]:
```suggestion
        assert: CLEAN_ == 0
        dirty_accumulator_[0] = CLEAN_
```

**Review by floitsch — APPROVED** [review]
LGTM.

File: src/pixel_display.toit [hunk 3]
```diff
@@ -272,27 +274,48 @@ abstract class PixelDisplay:
     finally:
       if not refreshed: refresh_ 0 0 0 0
 
-    bitmap_zap dirty_ CLEAN
+    bitmap_zap dirty_ CLEAN_
 
   // Clean determines if we should clean or draw the dirty area.
   update_frame_buffer clean/bool refresh_dimensions:
-    width := min width_ 128
+    width := min width_ 120
```
[the remark this answers is not in the document]
Comment (erikcorry) [reply src/pixel_display.toit:281]: 120x16 patches are more efficient than 128x8.  The package keeps the per-color byte arrays below 2000 bytes and this gives a size of 1920 bytes.

**Pull request merged**